src/diff.ts

```diff
@@ -368,8 +318,8 @@ function diffStyles(before, after) {
         const beforeLayers = [];
         if (before.layers) {
             before.layers.forEach((layer) => {
-                if (sourcesRemoved[layer.source]) {
-                    commands.push({command: operations.removeLayer, args: [layer.id]});
+                if (sourcesRemoved[(layer as any).source]) {
```

Comment:
Is the cast to `any` because not all layers have sources? Can you add a `'source' in layer` check first instead to make TypeScript happy without the cast?

Reply:
Sure, I'll see if this helps.

Reply:
Fixed.